### PR TITLE
GEOMESA-1171 Requiring explicit indication of mixed geometry types

### PR DESCRIPTION
**Pull request by elahrvivaz**
* Creating a schema with the default geometry type of 'Geometry' will throw an exception
* This can be overridden by including 'geomesa.mixed.geometries=true' in the feature type user data

Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/GeoMesaSchemaValidator.scala [hunk 1]
```diff
@@ -47,4 +56,30 @@ object TemporalIndexCheck extends LazyLogging {
...
+    if (gd != null && gd.getType.getBinding == classOf[Geometry]) {
+      val declared = sft.getUserData.get(MIXED_GEOMETRIES) match {
+        case null => false
+        case mixed: jBoolean if mixed => true
+        case mixed: String if jBoolean.valueOf(mixed) => true
+        case mixed if jBoolean.valueOf(mixed.toString) => true
+        case _ => false
+      }
+      if (!declared) {
+        throw new IllegalArgumentException("Trying to create a schema with mixed geometry type " +
```
Comment (anthonyccri) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/GeoMesaSchemaValidator.scala]: Nice

File: geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreTest.scala [hunk 2]
```diff
@@ -180,89 +191,51 @@ class AccumuloDataStoreTest extends Specification with AccumuloDataStoreDefaults
...
-        "records table" >> { c.tableOperations().exists(s"${defaultTable}_records") must beTrue }
-        "attr idx table" >> { c.tableOperations().exists(s"${defaultTable}_attr_idx") must beTrue }
+        val tables = GeoMesaTable.getTableNames(sft, ds)
+        tables must haveLength(4)
+        forall(Seq(Z2Table, Z3Table, AttributeTable, AttributeTable))(t => tables must contain(endWith(t.suffix)))
+        forall(tables)(t => ds.connector.tableOperations.exists(t) must beTrue)
       }
 
-      val pt = gf.createPoint(new Coordinate(0, 0))
+      val pt = WKTUtils.read("POINT (0 0)")
```
Comment (anthonyccri) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreTest.scala]: just wondering, why did you switch out the filter builder for text?

Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStoreTest.scala]: just trying to clean up the test... I find using ECQL and WKT is easier to grok, and you don't need filter factories and geometry factories floating around.

**Comment by anthonyccri** [comment]
Merged

**Pull request closed without merging**